x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -2142,6 +2212,130 @@ private static Expression typeSpecificConvert(ConvertFunction convert, Source so
         }
     }
 
+    private static class ResolveUnmapped extends ParameterizedAnalyzerRule<LogicalPlan, AnalyzerContext> {
+
+        @Override
+        protected LogicalPlan rule(LogicalPlan plan, AnalyzerContext context) {
+            return switch (context.unmappedResolution()) {
+                case UnmappedResolution.FAIL -> plan;
+                case UnmappedResolution.NULLIFY -> nullify(plan);
+                case UnmappedResolution.LOAD -> throw new IllegalArgumentException("unmapped fields resolution not yet supported");
+            };
+        }
+
+        private LogicalPlan nullify(LogicalPlan plan) {
```

Comment:
It's really unclear what to expect from this method without a short high-level explanation of the approach/algorithm in a javadoc.

Reply:
I've added further javadoc.